### ISSUE TITLE
Error: `new client: failed to run container: : fork/exec /usr/bin/docker: exec format error`

### ISSUE THREAD

**Issue by ceciliacsilva** [aweris/gale#119]
Hi. 

First: very interesting project 😊!

I'm having an error when trying to `gale run` (`gale list` works like a charm) even with this repo. I tried both ways with  `dagger run gale run ... ` and `gale run ...` and got the same results.

```shell
$ gale run .github/workflows/lint.yaml golangci-lint                       
Creating new Engine session... OK!
Establishing connection to Engine... 1: connect
1: > in init
1: starting engine 
1: starting engine [0.98s]
1: starting session 
1: starting session [0.28s]
1: connect DONE
OK!

6: resolve image config for ghcr.io/aweris/gale/tools/ghx:v0.0.7
6: > in from ghcr.io/aweris/gale/tools/ghx:v0.0.7
6: resolve image config for ghcr.io/aweris/gale/tools/ghx:v0.0.7 DONE

8: pull ghcr.io/aweris/gale/tools/ghx:v0.0.7
8: > in from ghcr.io/aweris/gale/tools/ghx:v0.0.7
8: resolve ghcr.io/aweris/gale/tools/ghx:v0.0.7@sha256:f4454a5b736fc05e6077fe88cee9b9d118089455234edb281822309cf7136c5a 
8: resolve ghcr.io/aweris/gale/tools/ghx:v0.0.7@sha256:f4454a5b736fc05e6077fe88cee9b9d118089455234edb281822309cf7136c5a [0.01s]
8: pull ghcr.io/aweris/gale/tools/ghx:v0.0.7 DONE

8: pull ghcr.io/aweris/gale/tools/ghx:v0.0.7 CACHED
8: > in from ghcr.io/aweris/gale/tools/ghx:v0.0.7
8: pull ghcr.io/aweris/gale/tools/ghx:v0.0.7 CACHED

13: upload . DONE
13: > in host.directory .
13: upload . DONE

13: upload .
13: > in host.directory .
13: transferring eyJvd25lcl9jbGllbnRfaWQiOiJ1eTdla3dheWo4M3QwcmdkYTA2bWc1YjBjIiwicGF0aCI6Ii4iLCJpbmNsdWRlX3BhdHRlcm5zIjpudWxsLCJleGNsdWRlX3BhdHRlcm5zIjpudWxsLCJmb2xsb3dfcGF0aHMiOm51bGwsInJlYWRfc2luZ2xlX2ZpbGVfb25seSI6ZmFsc2UsIm1heF9maWxlX3NpemUiOjB9: 
13: transferring eyJvd25lcl9jbGllbnRfaWQiOiJ1eTdla3dheWo4M3QwcmdkYTA2bWc1YjBjIiwicGF0aCI6Ii4iLCJpbmNsdWRlX3BhdHRlcm5zIjpudWxsLCJleGNsdWRlX3BhdHRlcm5zIjpudWxsLCJmb2xsb3dfcGF0aHMiOm51bGwsInJlYWRfc2luZ2xlX2ZpbGVfb25seSI6ZmFsc2UsIm1heF9maWxlX3NpemUiOjB9: 32.94MiB [0.23s]
13: upload . DONE

12: copy . CACHED
12: > in host.directory .
12: copy . CACHED

16: resolve image config for docker.io/alpine/git:latest
16: > in from alpine/git
16: resolve image config for docker.io/alpine/git:latest DONE

23: pull docker.io/alpine/git:latest
23: > in from alpine/git
23: resolve docker.io/alpine/git@sha256:b60414517a102de62fae6159fce9f1280892027469d21ddcf505b8d4139080b5 
23: resolve docker.io/alpine/git@sha256:b60414517a102de62fae6159fce9f1280892027469d21ddcf505b8d4139080b5 [0.01s]
23: pull docker.io/alpine/git:latest DONE

21: exec git rev-parse --symbolic-full-name HEAD CACHED
21: exec git rev-parse --symbolic-full-name HEAD CACHED

16: resolve image config for docker.io/alpine/git:latest
16: > in from alpine/git
16: resolve image config for docker.io/alpine/git:latest DONE

26: exec git rev-parse HEAD CACHED
26: exec git rev-parse HEAD CACHED

16: resolve image config for docker.io/alpine/git:latest
16: > in from alpine/git
16: resolve image config for docker.io/alpine/git:latest DONE

23: pull docker.io/alpine/git:latest CACHED
23: > in from alpine/git
23: pull docker.io/alpine/git:latest CACHED

43: exec git log -1 --follow --format=%H -- .github/workflows/lint.yaml
43: [0.17s] 89e207f49397b681b56666da1ffab3b9d81c0e0c
43: exec git log -1 --follow --format=%H -- .github/workflows/lint.yaml DONE

6: resolve image config for ghcr.io/aweris/gale/tools/ghx:v0.0.7
6: > in from ghcr.io/aweris/gale/tools/ghx:v0.0.7
6: resolve image config for ghcr.io/aweris/gale/tools/ghx:v0.0.7 DONE

64: resolve image config for ghcr.io/aweris/gale/services/artifact:v0.0.7
64: > in from ghcr.io/aweris/gale/services/artifact:v0.0.7
64: ...

63: resolve image config for ghcr.io/aweris/gale/services/artifactcache:v0.0.7 DONE
63: > in from ghcr.io/aweris/gale/services/artifactcache:v0.0.7
63: resolve image config for ghcr.io/aweris/gale/services/artifactcache:v0.0.7 DONE

64: resolve image config for ghcr.io/aweris/gale/services/artifact:v0.0.7 DONE
64: > in from ghcr.io/aweris/gale/services/artifact:v0.0.7
64: resolve image config for ghcr.io/aweris/gale/services/artifact:v0.0.7 DONE

78: resolve image config for ghcr.io/aweris/gale/runner/ubuntu:22.04
78: > in from ghcr.io/aweris/gale/runner/ubuntu:22.04
78: resolve image config for ghcr.io/aweris/gale/runner/ubuntu:22.04 DONE

135: pull ghcr.io/aweris/gale/services/artifactcache:v0.0.7
135: > in from ghcr.io/aweris/gale/services/artifactcache:v0.0.7
135: > in service cc9hc0oab4qtk.30g09cbj69b8k.dagger.local
135: resolve ghcr.io/aweris/gale/services/artifactcache:v0.0.7@sha256:20cc9a7822a0bf6401797c4b67f05f8e0bc4be196aec88a555aea7ff2d710ce9 
135: resolve ghcr.io/aweris/gale/services/artifactcache:v0.0.7@sha256:20cc9a7822a0bf6401797c4b67f05f8e0bc4be196aec88a555aea7ff2d710ce9 [0.01s]
135: pull ghcr.io/aweris/gale/services/artifactcache:v0.0.7 DONE

133: pull ghcr.io/aweris/gale/services/artifact:v0.0.7
133: > in from ghcr.io/aweris/gale/services/artifact:v0.0.7
133: > in service h94g2296tkk2u.30g09cbj69b8k.dagger.local
133: resolve ghcr.io/aweris/gale/services/artifact:v0.0.7@sha256:fef3d8d45d0b2a2c30ab0dca5cec149a0cbdb4a2f74923d63abbbf9099abdb1d 
133: resolve ghcr.io/aweris/gale/services/artifact:v0.0.7@sha256:fef3d8d45d0b2a2c30ab0dca5cec149a0cbdb4a2f74923d63abbbf9099abdb1d [0.01s]
133: pull ghcr.io/aweris/gale/services/artifact:v0.0.7 DONE

135: pull ghcr.io/aweris/gale/services/artifactcache:v0.0.7 CACHED
135: > in from ghcr.io/aweris/gale/services/artifactcache:v0.0.7
135: > in service cc9hc0oab4qtk.30g09cbj69b8k.dagger.local
135: pull ghcr.io/aweris/gale/services/artifactcache:v0.0.7 CACHED

133: pull ghcr.io/aweris/gale/services/artifact:v0.0.7 CACHED
133: > in from ghcr.io/aweris/gale/services/artifact:v0.0.7
133: > in service h94g2296tkk2u.30g09cbj69b8k.dagger.local
133: pull ghcr.io/aweris/gale/services/artifact:v0.0.7 CACHED

134: exec /entrypoint
134: > in service cc9hc0oab4qtk.30g09cbj69b8k.dagger.local
134: [0.14s] Starting server on port 8080
134: ...

139: mkdir / DONE
139: mkdir / DONE

139: mkdir / CACHED
139: mkdir / CACHED

144: pull ghcr.io/aweris/gale/runner/ubuntu:22.04
144: > in from ghcr.io/aweris/gale/runner/ubuntu:22.04
144: resolve ghcr.io/aweris/gale/runner/ubuntu:22.04@sha256:612d0fee8124ba5e09142023930c8b358af116de4ddcd236f57a88d302173584 
144: resolve ghcr.io/aweris/gale/runner/ubuntu:22.04@sha256:612d0fee8124ba5e09142023930c8b358af116de4ddcd236f57a88d302173584 [0.01s]
144: pull ghcr.io/aweris/gale/runner/ubuntu:22.04 DONE

138: mkfile /job_run.json
138: ...

143: copy /ghx /usr/local/bin/ghx CACHED
143: copy /ghx /usr/local/bin/ghx CACHED

141: mkdir /home/runner/_temp/ghx/runs/9
141: ...

138: mkfile /job_run.json DONE
138: mkfile /job_run.json DONE

137: copy / /home/runner/_temp/ghx/runs/9
137: ...

141: mkdir /home/runner/_temp/ghx/runs/9 DONE
141: mkdir /home/runner/_temp/ghx/runs/9 DONE

140: mkfile /home/runner/_temp/ghx/runs/9/event.json
140: ...

137: copy / /home/runner/_temp/ghx/runs/9 DONE
137: copy / /home/runner/_temp/ghx/runs/9 DONE

140: mkfile /home/runner/_temp/ghx/runs/9/event.json DONE
140: mkfile /home/runner/_temp/ghx/runs/9/event.json DONE

132: exec /entrypoint
132: > in service h94g2296tkk2u.30g09cbj69b8k.dagger.local
132: ...

145: exec /usr/local/bin/ghx run 9 ERROR: process "/usr/local/bin/ghx run 9" did not complete successfully: exit code: 1
145: [2.86s] Error: EOF: 1: connect
145: [2.86s] 1: > in init
145: [2.86s] 1: starting engine 
145: [2.86s] failed to list containers: fork/exec /usr/bin/docker: exec format error
145: [2.86s] 1: starting engine [1.05s]
145: [2.86s] 1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
145: [2.86s] Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
145: [2.86s] 
145: [2.86s] Please visit https://dagger.io/help#go for troubleshooting guidance.
145: [2.86s] Usage:
145: [2.86s]   ghx run <run-id> [flags]
145: [2.86s] 
145: [2.86s] Flags:
145: [2.86s]   -h, --help   help for run
145: [2.86s] 
145: [2.86s] Global Flags:
145: [2.86s]       --home string   home directory for ghx (default "/home/runner/_temp/ghx")
145: [2.86s] 
145: [2.86s] Error executing command: EOF: 1: connect
145: [2.86s] 1: > in init
145: [2.86s] 1: starting engine 
145: [2.86s] failed to list containers: fork/exec /usr/bin/docker: exec format error
145: [2.86s] 1: starting engine [1.05s]
145: [2.86s] 1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
145: [2.86s] Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
145: [2.86s] 
145: [2.86s] Please visit https://dagger.io/help#go for troubleshooting guidance.
145: exec /usr/local/bin/ghx run 9 ERROR: process "/usr/local/bin/ghx run 9" did not complete successfully: exit code: 1

132: exec /entrypoint
132: > in service h94g2296tkk2u.30g09cbj69b8k.dagger.local
Error: input:1: container.from.withUnixSocket.withFile.withEnvVariable.withMountedCache.withServiceBinding.withEnvVariable.withEnvVariable.withServiceBinding.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withMountedCache.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withMountedDirectory.withWorkdir.withSecretVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withNewFile.withEnvVariable.withMountedSecret.withDirectory.withExec.sync process "/usr/local/bin/ghx run 9" did not complete successfully: exit code: 1

Stdout:
Error executing command: EOF: 1: connect
1: > in init
1: starting engine 
failed to list containers: fork/exec /usr/bin/docker: exec format error
1: starting engine [1.05s]
1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error

Please visit https://dagger.io/help#go for troubleshooting guidance.
Stderr:
Error: EOF: 1: connect
1: > in init
1: starting engine 
failed to list containers: fork/exec /usr/bin/docker: exec format error
1: starting engine [1.05s]
1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error

Please visit https://dagger.io/help#go for troubleshooting guidance.
Usage:
  ghx run <run-id> [flags]

Flags:
  -h, --help   help for run

Global Flags:
      --home string   home directory for ghx (default "/home/runner/_temp/ghx")
Please visit https://dagger.io/help#go for troubleshooting guidance.
Error executing command: input:1: container.from.withUnixSocket.withFile.withEnvVariable.withMountedCache.withServiceBinding.withEnvVariable.withEnvVariable.withServiceBinding.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withMountedCache.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withMountedDirectory.withWorkdir.withSecretVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withEnvVariable.withNewFile.withEnvVariable.withMountedSecret.withDirectory.withExec.sync process "/usr/local/bin/ghx run 9" did not complete successfully: exit code: 1

Stdout:
Error executing command: EOF: 1: connect
1: > in init
1: starting engine 
failed to list containers: fork/exec /usr/bin/docker: exec format error
1: starting engine [1.05s]
1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error

Please visit https://dagger.io/help#go for troubleshooting guidance.
Stderr:
Error: EOF: 1: connect
1: > in init
1: starting engine 
failed to list containers: fork/exec /usr/bin/docker: exec format error
1: starting engine [1.05s]
1: connect ERROR: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error
Error: new client: failed to run container: : fork/exec /usr/bin/docker: exec format error

Please visit https://dagger.io/help#go for troubleshooting guidance.
Usage:
  ghx run <run-id> [flags]

Flags:
  -h, --help   help for run

Global Flags:
      --home string   home directory for ghx (default "/home/runner/_temp/ghx")
Please visit https://dagger.io/help#go for troubleshooting guidance.% 
```

- Platform: `Ubuntu 22.04.2 LTS`
- Gale version: 
```
{
  "major": "0",
  "minor": "0",
  "gitVersion": "v0.0.7",
  "gitCommit": "c5ea88e6a3fdf9c75d1923bda919365bfe4126de",
  "commitDate": "2023-09-04T12:48:02Z",
  "goVersion": "go1.20.7",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```
- docker engine version: `Docker version 24.0.6, build ed223bc`
- gh version: `gh version 2.34.0 (2023-09-06)`

**Comment by aweris** [comment]
> First: very interesting project 😊!

Thank you 🙏 

it's probably not able to find your docker socket. Could you try to run `gale validate .github/workflows/lint.yaml golangci-lint` and share command output?

**Comment by ceciliacsilva** [comment]
Yes. Logs:

```
arnold :: ~/external-projects/gale ‹main*› » gale validate .github/workflows/lint.yaml golangci-lint
[check] Dagger: PASSED
┏ 
┃ Global client is initialized
┃ Dagger runner host and/or session is not provided. Using docker socket instead.
┃ Docker socket is reachable
┃ Container execution is successful
┗ 
[check] GitHub CLI: PASSED
┏ 
┃ GitHub CLI is exist and configured properly
┗ 
[load] Repo: PASSED
┏ 
┃ Current repository gale is loaded
┗ 
[load] Workflow: PASSED
┏ 
┃ Workflows are loaded from .github/workflows
┃ Workflow .github/workflows/lint.yaml is loaded
┃ Job golangci-lint is loaded
┗ 
[check] Step Action: FAILED
┏ 
┃ [error] Node is not available in the runner image
┗ 
[check] Missing Expression Context: PASSED
[check] Docker Images: PASSED
┏ 
┃ All docker images are available
┗ 
[check] Run Step Shell: PASSED
```

**Comment by aweris** [comment]
Hmm, I need to investigate this when I have more time. Do you have any specific setup/config for `dagger` or `docker`?

**Comment by ceciliacsilva** [comment]
> Do you have any specific setup/config for dagger or docker?

I don't think so.

For the dagger-cli I followed the (vanilla) instructions described in https://docs.dagger.io/cli/465058/install (I didn't have any local version yet). For docker I had problems with version `23.x`, I updated to 24.0.6 before submitting this issue ([installing from packages](https://docs.docker.com/engine/install/ubuntu/#install-from-a-package)).

Thanks for trying to resolve this.
